### PR TITLE
Version 2 without config.toml with CLI

### LINKED ISSUE #10: CDA Test Patient Generator Implementation
We need to implement a generator for creating test patient data in CDA document format. This tool is crucial for improving the efficiency and accuracy testing of AKTIN components, particularly the CDA Importer.

**Current situation:**
* No structured method exists for generating test data that mirrors real clinical scenarios
* Testing quality is limited by lack of diverse, realistic test cases
* Difficult to identify potential weaknesses in import functionality before production

**Reference materials:**
* 6 example CDA documents ([storyboards.zip](https://github.com/user-attachments/files/17645971/storyboards.zip)) provided for reference
* [AKTIN CDA Guidelines](https://aktin.art-decor.pub/aktin-html-20180323T201638/index.html)
* [AKTIN Terminologies](https://aktin.art-decor.pub/aktin-html-20180323T201638/terminology.html)
* [AKTIN Templates](https://aktin.art-decor.pub/aktin-html-20180323T201638/rules.html)

**Goal:** Create a robust tool for generating realistic test patient data to improve testing capabilities and identify potential issues before production deployment.

### PR DESCRIPTION
**Pull request by sunberry00**

### CODE REVIEW

File: src/generator.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        Args:
+            count (int): The number of dates to generate.
+
+        Returns:
+            List[str]: A list of randomly generated date strings.
+        """
+        random_date = self.start_date + timedelta(
+            seconds=random.randint(0, int((self.end_date - self.start_date).total_seconds()))
+        )
+        if self.format == 'yyyymmdd':
```
Comment (sunberry00) [inline src/generator.py:71]: Sorry

File: src/generate_csv.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    excel_input = pd.read_excel(excel_path)
+
+    # Dictionary has form { conceptId -> (Default values, Type, Parameters, Nullable) }
+    variables_dict = extract_concept_id_attributes(excel_input)
+
+    # Parse parameters inside dictionary (Parameters now in dictionary format)
+    variables_dict = parse_parameters_to_dict(variables_dict)
+
+    # Generate data columns
+    output_data = generate_data_columns(variables_dict, num_datasets)
```
Comment (sunberry00) [inline src/generate_csv.py:124]: So

File: resources/value_sets/clinics.csv [hunk 3]
```diff
@@ -0,0 +1,78 @@
+klinik_name;city;postal_code
```
Comment (akomii) [inline resources/value_sets/clinics.csv]: clinic_name. Use ONE standard and keep it. Recheck also the other value_sets

File: resources/value_sets/individual_attributes.csv [hunk 4]
```diff
@@ -0,0 +1,101 @@
+vorname_m;vorname_f;nachname;strassenname
```
Comment (akomii) [inline resources/value_sets/individual_attributes.csv]: first_name_m;first_name_f;last_name;street_name

File: src/parser.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
+import os
+from datetime import datetime
+
+from config import config
+
+
+class Parser:
+    @staticmethod
+    def parse(string_to_parse):
```
Comment (akomii) [inline src/parser.py]: Add

```python
if not config_string or not isinstance(config_string, str):
    return {}
```

File: src/parser.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+            None
+        """
+        full_path = os.path.join(os.path.dirname(config.xlsx), value)
+        if os.path.isfile(full_path):
+            param_dict["link"] = full_path
+        else:
+            raise ValueError(f"File {value} does not exist in resources/value_sets")
+
+
+class DateHandler(AbstractHandler):
```
Comment (akomii) [inline src/parser.py]:
```python
try:
     params[self.key] = datetime.strptime(value, "%Y%m%d")
except ValueError:
     raise ValueError(f"Invalid date format. Expected YYYYMMDD, got: {value}")
```

File: src/parser.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
+import os
+from datetime import datetime
+
+from config import config
+
+
+class Parser:
```
Comment (akomii) [inline src/parser.py]: Rename to ConfigParser (because thats what this class is doing)

File: src/parser.py [hunk 8]
```diff
@@ -0,0 +1,176 @@
+import os
+from datetime import datetime
+
+from config import config
+
```
Comment (akomii) [inline src/parser.py]: Add error handling for malformed input strings in this script

File: src/value_remover.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import pandas as pd
+import numpy as np
+
+
+class ValueRemover:
+    @staticmethod
+    def remove_with_probability(column, probability):
```
Comment (akomii) [inline src/value_remover.py]: Input types? Return types?

Comment (akomii) [inline src/value_remover.py]: Better rename methods to `process_column` and `process_df`

File: src/value_remover.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        return column.mask(np.random.random(len(column)) < probability, '')
+
+    @classmethod
+    def remove_values_with_probability(cls, df, var_dict):
+        """
+        Remove values from the DataFrame based on the probabilities specified in var_dict.
+
+        Args:
+            df (pd.DataFrame): The DataFrame containing the data.
+            var_dict (dict): A dictionary where keys are concept IDs and values are tuples containing
```
Comment (akomii) [inline src/value_remover.py:27]: Give an example how var_dict can look like, i.e. `'col2': ('str', {}, False, 0.0)  # non-nullable, 0% null probability`

File: src/generator.py [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+        Generate random dates within the specified range and format.
+
+        Args:
+            count (int): The number of dates to generate.
+
+        Returns:
+            List[str]: A list of randomly generated date strings.
+        """
+        if self.format == 'yyyymmdd':
+            return [self._get_random_date().strftime('%Y%m%d') for _ in range(count)]
```
Comment (akomii) [inline src/generator.py]: Why are you parsing `yyyymmdd` to `%Y%m%d`. Why not directly using `%Y%m%d`?

File: src/generator.py [hunk 12]
```diff
@@ -0,0 +1,299 @@
...
+            column (str, optional): The column name in the CSV file to use for the value set. Defaults to None.
+        """
+        self.link = link
+        self.column = column
+        # TODO
+        self.dependent_columns = kwargs.get('dependent_column_1', None)
+        self.dependent_concept_id = kwargs.get('dependent_concept_id_1', None)
+        self._load_value_set_from_csv()
+
+    def _load_value_set_from_csv(self):
```
Comment (akomii) [inline src/generator.py:180]: What if the file is not found?

File: src/generate_csv.py [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+    Args:
+        excel_path (str): Path to the input Excel file.
+        csv_path (str): Path to the output CSV file.
+        num_datasets (int, optional): Number of datasets to generate. Defaults to 1.
+
+    Returns:
+        None
+    """
+    excel_input = pd.read_excel(excel_path)
```
Comment (akomii) [inline src/generate_csv.py:103]: Add validation of the excel columns

**Review by akomii — CHANGES_REQUESTED** [review]
Make sure that each method has the input/output data type. Recheck the docs. Less is more, but its good that they are so extensive

File: src/generator.py (hunk 12 above)
Comment (sunberry00) [reply src/generator.py:180]: It was already checked in parser but i add a new check here too

**Review by akomii — DISMISSED** [review]
╰(▔∀▔)╯

**Review by akomii — APPROVED** [review]

**Pull request merged**